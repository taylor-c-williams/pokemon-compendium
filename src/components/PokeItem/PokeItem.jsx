export default function PokeItem({
	pokemon,
	url_image,
	id,
	type_1,
	generation_id,
	hp,
	attack,
	defense,
	speed,
}) {
	return (
		<section className="pokemonCard">
			<img src={url_image} alt={pokemon} />
			<h1> {pokemon} </h1>
			Pokedex Number: {id}
			Type: {type_1}
			HP: {hp}
			ATK: {attack}
			DEF: {defense}
			SPD: {speed}
		</section>
	);
}
