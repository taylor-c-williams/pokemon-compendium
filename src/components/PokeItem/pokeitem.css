.pokeitem-body {
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	padding: 50px;
	font-style: italic;
	font-weight: 50;
	background-color: rgba(255, 255, 255, 0.9);
	border-radius: 300px;
	margin: 15px;
}

.stats {
	display: grid;
	grid-template-columns: 1fr 1fr;
	grid-template-rows: 1fr 1fr;
	grid-gap: 15px;
	margin-top: 0px;
}

p {
	margin: 10px;
}

img {
	height: 300px;
}
