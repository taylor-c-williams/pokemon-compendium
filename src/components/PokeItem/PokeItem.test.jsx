import { screen, render } from '@testing-library/react';
import PokeItem from './PokeItem';

it('accurately renders one pokemon object', () => {
	render(
		<PokeItem
			pokemon="bulbasaur"
			id="1"
			type_1="grass"
			attack="49"
			defense="49"
			hp="45"
			speed="45"
			url_image="http=//assets.pokemon.com/assets/cms2/img/pokedex/full/001.png"
		/>
	);
	const bulbasaur = screen.getByText('bulbasaur');
	expect(bulbasaur).toMatchSnapshot();
});
