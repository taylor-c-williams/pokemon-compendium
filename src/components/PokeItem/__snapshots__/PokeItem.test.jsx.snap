// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accurately renders one pokemon object 1`] = `
<h1>
   
  bulbasaur
   
</h1>
`;
